### PR TITLE
[Kubernetes] Retrieve node CPU/memory capacity from local information, not from kubelet /spec

### PR DESCRIPTION
**Pull request by astj**
#78 

Though I already submitted #79, this is another suggestion.

As described in title, now node CPU / memory capacity will be retrieved from local information, as ECS already does:
- Retrieve total memory of node from `/proc/meminfo`, via http://github.com/mackerelio/go-osstat
- Retrieve CPU cores by `runtime.NumCPU()`

Compared to #79, this change does not need extra privileges or configurations. So, existing users can safely migrate to this implementation.

Additionally, I propose some refinements:
- Place those logic into metric/hostinfo and share with both ECS and Kubernetes
- Remove platform.kubernetes.kubelet.client#GetSpec, since it's not used anymore

### CODE REVIEW

File: platform/ecs/metric_test.go [hunk 1]
```diff
@@ -60,7 +62,7 @@ func TestGenerateMetric(t *testing.T) {
 	for _, tc := range tests {
 		mock.metadataPath = fmt.Sprintf("taskmetadata/testdata/metadata_%s.json", tc.mode)
 		mock.statsPath = fmt.Sprintf("taskmetadata/testdata/stats_%s.json", tc.mode)
-		g := newMetricGenerator(mock)
+		g := newMetricGenerator(mock, hostinfo.NewMockGenerator(3876802560.0, 8.0, nil))
```
Comment (susisu) [inline platform/ecs/metric_test.go:65]: [note] these mock values can be any

File: platform/kubernetes/kubelet/client.go [hunk 2]
```diff
@@ -138,23 +136,6 @@ func (c *client) GetPodStats(ctx context.Context) (*kubeletTypes.PodStats, error
 	return stats, nil
 }
 
-// GetPodStats gets pod spec
-func (c *client) GetSpec(ctx context.Context) (*cadvisorTypes.MachineInfo, error) {
-	req, err := c.newRequest(specPath)
```
Comment (susisu) [inline platform/kubernetes/kubelet/client.go:143]: [nit] `specPath` is now unused

File: platform/kubernetes/kubelet/client_test.go [hunk 3]
```diff
@@ -227,7 +207,7 @@ func TestRequestToken(t *testing.T) {
 		t.Errorf("NewClient() should not raise error: %v", err)
 	}
 
-	_, err = c.GetSpec(ctx)
+	_, err = c.GetPod(ctx)
```
Comment (susisu) [inline platform/kubernetes/kubelet/client_test.go:210]: [note] This test checks that both `NewClient()` and internal `newRequest()` do not return error when given a non-empty token, and this method call can be anything that calls `newRequest()`.

**Review by susisu — COMMENTED** [review]
I haven't tested it yet, but LGTM

File: platform/kubernetes/kubelet/client.go (hunk 2 above)
Comment (astj) [reply platform/kubernetes/kubelet/client.go:143]: ! removed at https://github.com/mackerelio/mackerel-container-agent/pull/80/commits/4c8a6eedd2389e2017d9469212ec1195fa93230a

File: platform/ecs/metric_test.go (hunk 1 above)
Comment (astj) [reply platform/ecs/metric_test.go:65]: Yes. Actual values are not tested in ECS. On contrast they're tested in k8s, so maybe I can test them as k8s does. Do you think we should?

Comment (susisu) [reply platform/ecs/metric_test.go:65]: Basically it's better to improve test coverage, but I think ok for now.

**Review by susisu — APPROVED** [review]
:+1:

File: platform/ecs/metric_test.go (hunk 1 above)
Comment (astj) [reply platform/ecs/metric_test.go:110]: Okay, I added at https://github.com/mackerelio/mackerel-container-agent/pull/80/commits/fd37298937d4bd56e3a754e085b4778ef39e7c46 and it reveals the hostinfo is not used in current test cases 🧐

Furthermore, as far as I read https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html, CPU / memory limits are always present at least one of container-level or task-level, so this fallback is not necessary in ECS, but I'd like to keep this implementation for now……

**Comment by astj** [comment]
Thank you!

**Pull request merged**